rust/kernel/hrtimer.rs

```diff
@@ -0,0 +1,279 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Intrusive high resolution timers.
+//!
+//! Allows scheduling timer callbacks without doing allocations at the time of
+//! scheduling. For now, only one timer per type is allowed.
+//!
+//! # Example
+//!
+//! ```
+//!    use kernel::sync::Arc;
+//!    use kernel::{hrtimer::{RawTimer, Timer, TimerCallback}, impl_has_timer, pr_info, prelude::*, stack_pin_init};
+//!    use core::sync::atomic::AtomicBool;
+//!    use core::sync::atomic::Ordering;
```

Comment:
Rustfmt suggests:

```suggestion
use core::sync::atomic::AtomicBool;
use core::sync::atomic::Ordering;
use kernel::{
    hrtimer::{RawTimer, Timer, TimerCallback},
    impl_has_timer, pr_info,
    prelude::*,
    stack_pin_init,
};
```

Weird that it doesn't group the two `atomic` imports. `pr_info` should be in the prelude

Reply:
How did you get `rustfmt` to format the example for you? I tried with 

```shell
rustfmt +nightly --config format_code_in_doc_comments=true rust/kernel/hrtimer.rs 
```

but no success.